libcudacxx/include/cuda/std/__cccl/dialect.h

```diff
@@ -110,6 +104,13 @@
 #  define _CCCL_NO_VARIABLE_TEMPLATES
 #endif // _CCCL_STD_VER <= 2011
 
+// Variable templates are more efficient most of the time, so we want to use them rather than structs when possible
+#if defined(_CCCL_NO_VARIABLE_TEMPLATES)
```

Comment:
This actually relaxes the condition to C++14 and sufficient support for variable templates.

Should we stay with support for inline variables

Reply:
If `_CCCL_TRAIT` is also use to refer to std traits, then this won't work because variable templates are available in C++14 but all the variable templates for std traits were added in C++17.

When is `_CCCL_TRAIT` used? Only for stuff in `::cuda::std`? Then it should be fine.

Reply:
That is one of the reason we are using our internal traits all the time ;)

Reply:
I mean I am open to keep it as is and use `_CCCL_HAS_NO_INLINE_VARIABLES` as the condition, however, I believe we can do better

Reply:
If we only use it for the internal traits, then all is good! I just wondered whether we would cause any breakage.